scripts/resource/class_generator.py

```diff
@@ -0,0 +1,328 @@
+from __future__ import annotations
+import shlex
+import os
+
+from typing import Any, Dict, List, Optional, Tuple
+import click
+import re
+
+from pyhelper_utils.shell import run_command
+import yaml
+
+from ocp_resources.resource import Resource
+
+from jinja2 import DebugUndefined, Environment, FileSystemLoader, meta
+from simple_logger.logger import get_logger
+
+TYPE_MAPPING: Dict[str, str] = {
+    "<integer>": "int",
+    "<Object>": "Dict[Any, Any]",
+    "<[]Object>": "List[Any]",
+    "<string>": "str",
+    "<map[string]string>": "Dict[Any, Any]",
+    "<boolean>": "bool",
+}
+LOGGER = get_logger(name="class_generator")
+
+
+def get_oc_or_kubectl() -> str:
+    if os.system("which oc") == 0:
+        return "oc"
+
+    elif os.system("which kubectl") == 0:
+        return "kubectl"
+
+    else:
+        LOGGER.error("oc or kubectl not available")
+        return ""
+
+
+def check_cluster_available() -> bool:
+    _exec = get_oc_or_kubectl()
+    if not _exec:
+        return False
+
+    return os.system(f"{_exec} version") == 0
+
+
+def get_kind_data(kind: str) -> Dict[str, Any]:
+    """
+    Get oc/kubectl explain output for given kind and if kind is namespaced
+    """
+    if not check_cluster_available():
+        LOGGER.error("Cluster not available")
+        return {}
+
+    explain_rc, explain_out, explain_err = run_command(command=shlex.split(f"oc explain {kind} --recursive"))
```

Comment:
why is it needed if `--file` is passed by the user?

Reply:
I don't understand the question, User can pass file or kind.

Reply:
okay I see `Either pass --file or --kind (When passing --kind a working cluster is required)`, thanks